lib/dockerregistry/checks.go

```diff
@@ -0,0 +1,159 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inventory
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+
+	gogit "gopkg.in/src-d/go-git.v4"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+)
+
+// ImageRemovalCheck implements the PreCheck interface and checks against
+// pull requests that attempt to remove any images from the promoter manifests.
+type ImageRemovalCheck struct {
+	CurrentGitRepo url.URL
+	MasterSHA      plumbing.Hash
+	PullRequestSHA plumbing.Hash
+}
+
+// MKRealImageRemovalCheck returns an instance of ImageRemovalCheck.
+func MKRealImageRemovalCheck(
+	currentGitRepoStr string,
+) (*ImageRemovalCheck, error) {
+	currentGitRepo, err := url.Parse(currentGitRepoStr)
+	if err != nil {
+		return nil, fmt.Errorf("Could not parse the git repo url: %v", err)
+	}
+	masterSHAStr := os.Getenv("PULL_BASE_SHA")
+	if masterSHAStr == "" {
+		return nil, fmt.Errorf("The PULL_BASE_SHA environment variable " +
+			"doesn't exist")
```

Comment:
WRT "doesn't exist" --- this is not always true. It could be that the environment variable is set to the 0-length (empty) string. Easier to just say that it is invalid.